### ISSUE TITLE
Wrong LCOM metric when using promoted properties in constructor 

### ISSUE THREAD

**Issue by omidulus** [phpmetrics/PhpMetrics#485]
# Bug report
v2.8.1

When using constructor promoted properties the application is unable to establish a relationship between the __construct method and other class method, thus artificially increasing LCOM metric.

Example:
LCOM here is **2** but should be **1**.

```
<?php
......

class ExpectationExecutionMessageHandler
{
    public function __construct(
        public ExecutorService $executorService
    ) {
    }

    /** @throws ExceptionInterface */
    public function __invoke(DispatchExpectation $dispatchExpectation): void
    {
        $this->executorService->execute($dispatchExpectation->getExpectationUuid());
    }
}

```

**Comment by niconoe-** [comment]
This is fixed in the incoming version v3. You can give it a try via composer installation using `composer require phpmetrics/phpmetrics:dev-3.x-dev`

**Comment by niconoe-** [comment]
@omidulus The v3.0.0rc1 has just been released. Please give it a try and share your feedback regarding this issue, as it should now be solved.

Thanks a lot!

**Comment by niconoe-** [comment]
Fixed since v3.0.0rc1. Closing.
